src/test/java/com/PlaceLab/Amila/qamp/pages/SinglePlaceSearchPage.java

```diff
@@ -0,0 +1,116 @@
+package com.PlaceLab.Amila.qamp.pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+
+import java.time.Duration;
+
+public class SinglePlaceSearchPage {
+    private final static By SPS_NAV_BAR = By.xpath("/html/body/div[2]");
+    private final static String SPS_EXPECTED_URL = "https://demo.placelab.com/dashboard/traffic";
+    private final static By SPS_DROPDOWN_MENI_CLICK = By.xpath("//*[@id=\"create-menu\"]/i");
+    private final static By SPS_DROPDOWN_MENI_OPEN = By.xpath("/html/body/div[2]/div[2]");
+    private final static By TITLE_PLACES = By.xpath("/html/body/div[2]/div[2]/div/div[2]/h6");
+    private final static By SINGLE_PLACE_SEARCH_FIELD_DROPDOWN_MENI = By.cssSelector("#singleplacesearch");
+    private final static By SINGLE_PLACE_SEARCH_BTN = By.cssSelector("#singleplacesearch");
+    private final static String SPS_EXPECTED_URL_1 = "https://demo.placelab.com/places/single_place_searches/new";
+    private final static By SPS_PAGE_HEADER = By.cssSelector("#single_poi_dialog > div.create-report-header > div");
+    private final static By VISIBLE_REPORT_CONTENT = By.cssSelector("#sp_query");
+    private final static By REQUIRED_FIELDS_MARK = By.cssSelector("#single_poi_dialog > div.required-fields-notice > span");
+    private final static By USAGE_PIE_CALCULATOR = By.className("usage-calculator");
+    private final static By CREATE_SPS_REPORT_BTN = By.cssSelector("button.btn.large-btn.run-btn.run-all-btn");
+    private final static By REPORT_NAME = By.cssSelector("#name");
+    private final static By PLACE_NAME = By.cssSelector("#single_text");
+    private final static By CREATE_REPORT_BTN_DISABLED = By.cssSelector("button.btn.large-btn.run-btn.run-all-btn[disabled='disabled']");
+    private final static By LOCATION_NAME = By.cssSelector("#location_name");
+    private final static By AUTOCOMPLETE_DROPDOWN = By.cssSelector(".typeahead.dropdown-menu");
+    private final static By POP_UP = By.xpath("/html/body/div[10]/div[3]/div/button[1]");
+    private final WebDriver driver;
+    private LoginPage loginPage;
+
+    public SinglePlaceSearchPage(final WebDriver webDriver) {
+        this.driver = webDriver;
+    }
+
+    public void validateSinglePlacePageContent() {
+
+        String actualUrl = driver.getCurrentUrl();
+        Assert.assertEquals(actualUrl, SPS_EXPECTED_URL, "Verify dashboard traffic page successfully opened.");
+        Assert.assertTrue(driver.findElement(SPS_NAV_BAR).isDisplayed(), "Verify if Navigation bar is visible.");
+
+        driver.findElement(SPS_DROPDOWN_MENI_CLICK).click();
+
+        try {
+            Thread.sleep(200);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
instead of sleeps, use selenium waits everywhere

Reply:
Done.